fdbrpc/fdbrpc.h

```diff
@@ -353,20 +354,29 @@ struct NetNotifiedQueueWithAcknowledgements final : NotifiedQueue<T>,
 				acknowledgements = AcknowledgementReceiver(
 				    FlowTransport::transport().loadedEndpoint(message.get().asUnderlyingType().acknowledgeToken.get()));
 			}
-			if (this->shouldFireImmediately()) {
-				// This message is going to be consumed by the client immediately (and therefore will not call pop()) so
-				// send an ack immediately
-				if (acknowledgements.getRawEndpoint().isValid()) {
-					acknowledgements.bytesAcknowledged += message.get().asUnderlyingType().expectedSize();
-					FlowTransport::transport().sendUnreliable(
-					    SerializeSource<ErrorOr<AcknowledgementReply>>(
-					        AcknowledgementReply(acknowledgements.bytesAcknowledged)),
-					    acknowledgements.getEndpoint(TaskPriority::ReadSocket),
-					    false);
+			if (acknowledgements.sequence != message.get().asUnderlyingType().sequence) {
+				TraceEvent(SevError, "StreamSequenceMismatch")
+				    .detail("Expected", acknowledgements.sequence)
+				    .detail("Actual", message.get().asUnderlyingType().sequence);
+				ASSERT_WE_THINK(false);
```

Comment:
I am a bit confused about this one: I thought the simulation failure did exactly produce this? Is there something else that I am missing which prevents this from happening?